posthog/hogql_queries/experiments/test/test_experiment_trends_query_runner.py

```diff
@@ -846,6 +867,25 @@ def test_query_runner_with_data_warehouse_series_no_end_date_and_nested_id(self)
                     timestamp=datetime(2023, 1, i + 1),
                 )
 
+        _create_person(
+            team=self.team,
+            distinct_ids=["internal_test_1"],
+            properties={"email": "internal_test_1@posthog.com", "$user_id": "internal_test_1"},
+        )
+
+        _create_event(
+            team=self.team,
+            event="$feature_flag_called",
+            distinct_id="internal_test_1",
+            properties={
+                feature_flag_property: "test",
+                "$feature_flag_response": "test",
+                "$feature_flag": feature_flag.key,
+                "$user_id": "internal_test_1",
+            },
+            timestamp=datetime(2023, 1, 3),
+        )
```

Comment:
I’d prefer a clearer test to explicitly verify that this event is excluded from the returned data. It might be better to have a separate method specifically for testing the filtering of test users.

Reply:
How might you suggest constructing the test?

Reply:
Something like:
- Only populate the data warehouse records with rows including the test user id
- The experiment query should return no result

My point is: we’re creating a test user and an event generated by them. Am I missing something, or what exactly are we doing with this user and event later on? How are we ensuring that this event is excluded from the results?

Reply:
> How are we ensuring that this event is excluded from the results?

It's inferred. These values will change significantly if the event is included in results:

https://github.com/PostHog/posthog/pull/26911/files#diff-36098f224702c73f16f7fce4f65403a8d9f840e079324042dc4dabeac1176f4dL924-R971

The way I went about writing the test was:
1. Add the data warehouse row and event so the test was failing.
2. Add the persons table join and `filterTestAccounts=True` statement so the test is passing with the original conditions again.

I think what you suggest is marginally more explicit, but could also be unexpectedly erroneous in some other ways.

I updated the test to also run the query with `filterTestAccounts=false` and assert the results: 5ca3fd30d6

Reply:
Sounds good 👍